contracts/SimpleSwap.sol

```diff
@@ -65,126 +56,66 @@ contract SimpleSwap {
+      hardDeposits[beneficiary].amount = hardDeposits[beneficiary].amount.sub(hardDepositUsage);
       totalHardDeposit = totalHardDeposit.sub(hardDepositUsage);
     }
     /* increase the stored paidOut amount to avoid double payout */
-    cheque.paidOut = cheque.paidOut.add(totalPayout);
+    paidOutCheques[beneficiary] = paidOutCheques[beneficiary].add(totalPayout);
     /* do the actual payments */
-    beneficiaryAgent.transfer(totalPayout.sub(calleePayout));
-    emit ChequeCashed(beneficiaryPrincipal, beneficiaryAgent, msg.sender, cheque.serial, totalPayout, requestPayout, calleePayout);
+    recipient.transfer(totalPayout.sub(calleePayout));
+    /* do a transfer to the callee if specified*/
+    if(calleePayout != 0) {
+      msg.sender.transfer(calleePayout);
+    }
+    emit ChequeCashed(beneficiary, recipient, msg.sender, totalPayout, cumulativePayout, calleePayout);
+    /* let the world know that the issuer has over-promised on outstanding cheques */
     if(requestPayout != totalPayout) {
       emit ChequeBounced();
     }
   }
 
   function cashCheque(
-    address beneficiaryPrincipal,
-    address payable beneficiaryAgent,
-    uint requestPayout,
+    address beneficiary,
+    address payable recipient,
+    uint cumulativePayout,
     bytes memory beneficiarySig,
-    uint256 expiry,
-    uint256 calleePayout
+    uint256 calleePayout,
+    bytes memory issuerSig
   ) public {
-    require(now <= expiry, "SimpleSwap: beneficiarySig expired");
-    require(beneficiaryPrincipal == recover(cashOutHash(
+    /* the beneficiary must have given explicit approval for cashing out the cumulativePayout and the calleePayout payment */
+    require(beneficiary == recover(cashOutHash(
       address(this),
       msg.sender,
-      requestPayout,
-      beneficiaryAgent,
-      expiry,
+      cumulativePayout,
+      recipient,
       calleePayout
       ), beneficiarySig), "SimpleSwap: invalid beneficiarySig");
-    _cashChequeInternal(beneficiaryPrincipal, beneficiaryAgent, requestPayout, calleePayout);
-    msg.sender.transfer(calleePayout);
+    _cashChequeInternal(beneficiary, recipient, cumulativePayout, calleePayout, issuerSig);
   }
-  /// @notice attempt to cash latest chequebeneficiary
-  /// @param beneficiaryAgent agent (of the beneficiary) who receives the payment (i.e. other chequebook contract or the beneficiary)
-  /// @param requestPayout amount requested to pay out
-  function cashChequeBeneficiary(address payable beneficiaryAgent, uint requestPayout) public {
-    _cashChequeInternal(msg.sender, beneficiaryAgent, requestPayout, 0);
+  /// @notice Cash a cheque
+  /// @param recipient agent (of the beneficiary) who receives the payment (i.e. other chequebook contract or the beneficiary)
+  /// @param cumulativePayout amount requested to pay out
+  function cashChequeBeneficiary(address payable recipient, uint cumulativePayout, bytes memory issuerSig) public {
```

Comment:
Since we no longer have timeouts anyone should be able to call this now (for any beneficiary). In that case `recipient` should be always the same as `beneficiary`. If a custom `recipient` is needed the other function should be used.

Reply:
Why do you think anyone should always be able to call this function for any beneficiary? I don't see what we would gain from this, but it introduces edge-cases on the side implementing a chequebook, as now somebody can cash out a cheque of a `beneficiary`, without his consent.

Reply:
Anyone be able to submit cheques was part of the original Swap design. It was only removed because of the issues it caused together with the timeout mechanism. I don't think it's really necessary but I remember @zelig and @nagydani wanting this.

Reply:
I see. I believe, however, that the current design is easier to work as nobody will be able to submit a cheque without explicit consent from the beneficiary. If a different party wants to submit, he can do so via the `submitCheque` function, if he has the signature of the beneficiary. Furthermore, the `submitCheque` function allows the beneficiary to compensate the callee for the gas-costs.

Reply:
To provide a summary:
- Removing the `recipient` and always sending the money to the `beneficiary` will allow anybody to be able to call this function. In the current implementation, smart-contracts without the possibility to call `cashChequeBeneficiary` will *not* be able to cash cheques.
- Leaving the code as-is gives more control to the `beneficiary` as nobody will be able to cash a cheque without his consent. If a beneficiary wants to send money to a smart-contract on cash-out, the smart contract-contract can be a `recipient`. However, no guarantees can be given on a particular smart-contract always/ever being the recipient.